ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezSessionState.java

```diff
@@ -350,6 +351,16 @@ protected void openInternal(String[] additionalFilesNotFromConf,
 
     setupSessionAcls(tezConfig, conf);
 
+    /*
+     * Update HADOOP_CREDSTORE_PASSWORD for the TezAM.
+     * If there is a job specific credential store, it will be set.
+     * HiveConfUtil.updateJobCredentialProviders should not be used here,
+     * as it changes the credential store path too, which causes the dag submission fail,
+     * as this config has an effect in HS2 (on TezClient codepath), and the original hadoop
+     * credential store should be used.
+     */
+    HiveConfUtil.updateCredentialProviderPasswordForJobs(tezConfig);
```

Comment:
Do you know if that information won't be localized by Tez in plain text?

Reply:
the credential store password appears in the launch-container.sh script, which is created by yarn, and it contains the environment variables, I think this should be the same as in case of any other execution engines that's localized by yarn, but I'm sure that launch-container.sh is not included into application logs, so it's only readable of somebody having access to the cluster nodes